openhands/controller/agent_controller.py

```diff
@@ -219,9 +223,18 @@ async def on_event(self, event: Event):
         if hasattr(event, 'hidden') and event.hidden:
             return
 
-        # if the event is not filtered out, add it to the history
-        if not any(isinstance(event, filter_type) for filter_type in self.filter_out):
```

Comment:
Is this filter no longer needed?

Reply:
@enyst wdyt about this?

Reply:
You're right, it's just that certain kinds of events shouldn't be in the agent history (like "backend stuff", the LLM doesn't need to know that)

Reply:
So it should still be there then or not? 😄

Reply:
We should continue to filter.